docs/scanners/gitleaks.md

```diff
@@ -0,0 +1,43 @@
+# [Gitleaks](https://github.com/zricethezav/gitleaks)
+
+The [Gitleaks Scanner](https://github.com/zricethezav/gitleaks) tool for finding secrets within a directory or Git repo.
+
+## Configuration
+
+Gitleaks currently always runs with the `--recursive` and `--format json` options turned on.
+In addition, you can configure the additional options below.
+For more information on Gitleaks configuration, see the docs [here](https://github.com/zricethezav/gitleaks#usage-and-options)
+
+```yaml
+  scanner_configs:
+    Bandit:
+      config-path: "./gitleaks.conf"
+      repo-config-path: "./gitleaks.conf"
+      threads: 1
+      unstaged: true
+      branch: "main"
+      redact: true
+      no-git: true
+      files-at-commit: "aaaaaaaaaaa"
+      commit: "aaaaaaaaaaa"
+      commits:
+      - "aaaaaaaaaaa"
+      - "aaaaaaaaaaa"
+      commits-file: "./commits.txt"
+      commit-from: "aaaaaaaaaaa"
+      commit-to: "aaaaaaaaaaa"
+      commit-since: "2020-01-01T00:00:00-0000"
+      commit-until: "2020-01-01T00:00:00-0000"
+      depth: 1
+```
```

Comment:
Do your test cases cover each one of the config options here?
What about some combinations of the options?

Reply:
All config options other than `threads` are tested. A number of the `commit*` options are tested together.